### DISCUSSION TITLE
Unexpected Behavior with VLAN Group Scoping and Virtual Machines

### DISCUSSION THREAD

**Discussion by adparis99** [netbox-community/netbox#11374] [Q&A]
Hello!

I've been using NetBox for a while and feel pretty comfortable with a lot of it. One of the things I've been trying to do recently is organize our VLANs into VLAN Groups. This has worked nearly perfectly. However, I've run into a strange issue. The following is my criteria/issue recreation:

1. Create a VLAN Group (Group X) scoped to Site X, Location X
2. Create a VLAN (VLAN X) assigned to Group X
3. Create a virtual host device (Device X) that is assigned to a cluster, site (Site X), and location (Location X)
4. Create a virtual machine (VM X) assigned to Device X
5. Add an interface to VM X
6. Attempt to assign VLAN X to the newly created interface on VM X

The problem occurs here, as VLAN X doesn't show up in the list of VLANs available to be assigned to the virtual machine interface. Other VLANs that are in groups scoped directly to Site X as well as those not yet in groups show up as available, but not any VLANs assigned to Location X. From what I can tell, this is not the case with devices. For example, VLAN X is available to be assigned to interfaces belonging to Device X. 

If I had to guess, it seems like because VMs can only be assigned to sites and not locations, only the VLANs scoped to or otherwise available to the site assigned to the VM are able to be selected. But given that VM X in this case is assigned to a device that is assigned to a specific location, I assumed that VLANs associated with Location X would also be available for assignment. Is that the incorrect assumption, or is this a bug? If the former, what might you guys suggest in this case as a workaround?

**Comment by joaolucasmacedo** [comment]
I've had the same issue.

**Comment by tacerus** [comment]
We encounter this as well, did you find a solution or should we turn it into an issue?
